content/guides/advanced/testing-with-jest-and-bankrun.md

```diff
@@ -0,0 +1,380 @@
+   you get an error saying that you can not log after the test is finished your
+   test is probably timing out. Different to Mocha Jest does not have a default
+   timeout. You can set a timeout in your `jest.config.js` file:
+
+```js
+module.exports = {
+  testTimeout: 10000,
+};
+```
+
+Or you can set a timeout for a single test:
+
+```js
+test("test name", async () => {
+  // your test code
+}, 10000);
+```
+
+3. If you get an error saying that Anchor environment variables are missing you
+   are probably trying to use the AnchorProvider without running the tests
+   through `anchor test` or `anchor run test`. Just update your Anchor.toml to
+   run the Jest tests `yarn test` to run them in the anchor environment with all
+   the environment variables set.
+
+## Bankrun
+
+Instead of using `solana-test-validator` you can also use
+[Solana Bankrun](https://kevinheavey.github.io/solana-bankrun/). It acts
+similarly to the local validator but is more lightweight and faster. Some really
+helpful features are
+[writing custom account data](https://kevinheavey.github.io/solana-bankrun/tutorial/#writing-arbitrary-accounts)
+and
+[time travel](https://kevinheavey.github.io/solana-bankrun/tutorial/#time-travel)
+which makes it much easier to test time based programs and programs that rely on
+specific account data.
+
+To use Bankrun and Bankrun Anchor you need to add it to your `package.json`:
+
+```bash
+yarn add solana-bankrun anchor-bankrun
+```
+
+To switch from a Mocha Anchor test to a
+[Bankrun Anchor](https://github.com/kevinheavey/anchor-bankrun) test you only
+need to change the provider to be a
+[`BankrunProvider`](https://kevinheavey.github.io/solana-bankrun/tutorial/#anchor-integration)
+and create a context using `startAnchor`:
+
+```js
+const context = await startAnchor(".", [], []);
+const client = context.banksClient;
+
+const provider = new BankrunProvider(context);
+anchor.setProvider(provider);
+```
+
+`startAnchor` will automatically add your program from your Anchor workspace to
+the Bankrun bank. You can also
+[add additional accounts and programs](https://kevinheavey.github.io/solana-bankrun/tutorial/#writing-arbitrary-accounts)
+to the bank by passing them in the `startAnchor` function. There are a few
+things that are different to running tests against the local validator though
+which we will cover in the next section.
+
+### Bankrun differences to the local validator
+
+Because Bankrun directly runs your tests against a local Bank the tests are
+faster, but there are a few differences to the local validator:
+
+#### Airdrops
+
+- Bankrun does not support Airdrops. The standard signer used in the
+  BankrunProvider will be automatically funded with some Sol. If you need
+  another funded account you can create one by passing in an additional account
+  in the `startAnchor` function.
+
+```js
+let secondKeypair: Keypair = new anchor.web3.Keypair();
+
+context = await startAnchor(
```

Comment:
why does some of the example code use `startAnchor` and some uses `start`?
the bankrun docs page's example use `startAnchor` but it looks like many of the bankrun tests we have in program-examples uses `start`

whats the difference? we should stick to one within this guide (and maybe add a callout somewhere stating the differences, or that they are the same)

Reply:
start is for native programs and startAnchor for anchor programs. Its explained a little further down as "Using bankrun for native programs".